### PR TITLE
Fix edge case where retry wait could be negative

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

File: httpclient/src/main/java/com/okta/sdk/impl/http/httpclient/HttpClientRequestExecutor.java [hunk 1]
```diff
@@ -429,21 +429,32 @@ private URI getRedirectUri(HttpResponse httpResponse) {
...
 
-        // default / fallback strategy
+        // default / fallback strategy (backwards compatible implementation)
         if (delay < 0) {
             delay = Math.min(getDefaultDelayMillis(retries), timeElapsedLeft);
         }
 
+        // this shouldn't happen, but guard against a negative delay at this point
+        if (delay < 0) {
+            throw failedToRetry();
```
Comment (nbarbettini) [inline httpclient/src/main/java/com/okta/sdk/impl/http/httpclient/HttpClientRequestExecutor.java]: I'm assuming this means the call fails?

Comment (bdemers) [reply httpclient/src/main/java/com/okta/sdk/impl/http/httpclient/HttpClientRequestExecutor.java]: correct the `failedToRetry()` method just returns a new exception

File: httpclient/src/main/java/com/okta/sdk/impl/http/httpclient/HttpClientRequestExecutor.java [hunk 2]
```diff
@@ -454,6 +465,10 @@ private void pauseBeforeRetry(int retries, HttpResponse httpResponse, long timeE
         }
     }
 
+    private RestException failedToRetry() {
+        return new RestException("Cannot retry request, next request will exceed retry configuration.");
```
Comment (bretterer) [inline httpclient/src/main/java/com/okta/sdk/impl/http/httpclient/HttpClientRequestExecutor.java]: May be nice to give some more context here....

"Cannot retry request, next request will exceed the retry configuration of XX.  You have been retrying for XX.

Not a blocker though

File: httpclient/src/test/groovy/com/okta/sdk/impl/http/httpclient/HttpClientRequestExecutorTest.groovy [hunk 3]
```diff
@@ -277,13 +278,26 @@ class HttpClientRequestExecutorTest {
 
         // maxElapsed enabled
         requestExecutor.maxRetries = 0
-        requestExecutor.maxElapsedMillis = 15000
+        requestExecutor.maxElapsedMillis = 15000L
```
Comment (bretterer) [inline httpclient/src/test/groovy/com/okta/sdk/impl/http/httpclient/HttpClientRequestExecutorTest.groovy]: is the `L` a typo?

**Pull request merged**